lib/js/src/rpc/structs/VideoStreamingFormat.js

```diff
@@ -31,38 +31,54 @@
 */
 
 import { RpcStruct } from '../RpcStruct.js';
+import { VideoStreamingProtocol } from '../enums/VideoStreamingProtocol.js';
+import { VideoStreamingCodec } from '../enums/VideoStreamingCodec.js';
 
 class VideoStreamingFormat extends RpcStruct {
+    /**
+    * @constructor
+    */
+    constructor () {
+        super();
+    }
     /**
     * @param {VideoStreamingProtocol} val
     * @return {VideoStreamingFormat}
     */
     setProtocol (val) {
-        this.setParameter(VideoStreamingFormat.KEY_PROTOCOL, val);
-        return this;
+        const errorMessage = 'val must be an instance of VideoStreamingProtocol';
+        if (val instanceof VideoStreamingProtocol) {
+            this.setParameter(VideoStreamingFormat.KEY_PROTOCOL, val);
+            return this;
+        }
+        throw new Error(errorMessage);
```

Comment:
Validation should be done via the `validateType` method provided in the base `RpcStruct` class.

Reply:
fixing